test/e2e/scalenodes.go

```diff
@@ -101,31 +104,52 @@ var _ = Describe("Scale nodes", func() {
 			err = waitForScale(mss.Items[1].Name)
 			Expect(err).NotTo(HaveOccurred())
 
+			replicas, err = countReplicas()
+			Expect(err).NotTo(HaveOccurred())
+			Expect(replicas).To(BeEquivalentTo(minSupportedReplicas))
+
+			ms, err := clients.MachineAPI.MachineV1beta1().MachineSets(machineSetsNamespace).List(ctx, metav1.ListOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(mss.Items).NotTo(BeEmpty())
+
+			// Find which machineset is still zeroed out, and re-add the replica
+			for _, machineset := range ms.Items {
+				if machineset.Status.Replicas != 1 {
+					err = scale(machineset.Name, 1)
+					Expect(err).NotTo(HaveOccurred())
+
+					err = waitForScale(machineset.Name)
+					Expect(err).NotTo(HaveOccurred())
+				}
+			}
+
 		case len(mss.Items) < 3:
-			// E2E cluster has no AZs, remove two replicas from 1 machineset
+			// Region has no availability zones, remove two replicas from the single machineset
 			err = scale(mss.Items[0].Name, -2)
 			Expect(err).NotTo(HaveOccurred())
 
 			err = waitForScale(mss.Items[0].Name)
 			Expect(err).NotTo(HaveOccurred())
-		}
 
-		ms, err := clients.MachineAPI.MachineV1beta1().MachineSets(machineSetsNamespace).List(context.Background(), metav1.ListOptions{})
-		Expect(err).NotTo(HaveOccurred())
+			replicas, err = countReplicas()
+			Expect(err).NotTo(HaveOccurred())
+			Expect(replicas).To(BeEquivalentTo(minSupportedReplicas))
 
-		replicaCount := 0
-		for _, machineset := range ms.Items {
-			if machineset.Spec.Replicas != nil {
-				replicaCount += int(*machineset.Spec.Replicas)
-			}
+			// Re-add the remaining missing replica
+			err = scale(mss.Items[0].Name, 1)
```

Comment:
Similar to the multi-zone test, assumes a starting state of 2+1 replicas.  Stash the actual account then use that to set the desired state here.

Reply:
@jewzaam let me know what you think of the recent changes (also addressed @petrkotas comment).